packages/dependency-extraction-webpack-plugin/lib/index.js

```diff
@@ -163,127 +163,104 @@ class DependencyExtractionWebpackPlugin {
 
 		const combinedAssetsData = {};
 
-		// Process each entry point independently.
-		for ( const [
-			entrypointName,
-			entrypoint,
-		] of compilation.entrypoints.entries() ) {
-			const entrypointExternalizedWpDeps = new Set();
+		// Accumulate all entrypoint chunks, some of them shared
+		const entrypointChunks = new Set();
+		for ( const entrypoint of compilation.entrypoints.values() ) {
+			for ( const chunk of entrypoint.chunks ) {
+				entrypointChunks.add( chunk );
+			}
+		}
+
+		// Process each entrypoint chunk independently
+		for ( const chunk of entrypointChunks ) {
+			const chunkFiles = Array.from( chunk.files );
+
+			const chunkJSFile = chunkFiles.find( ( f ) => /\.js$/i.test( f ) );
+			if ( ! chunkJSFile ) {
+				// There's no JS file in this chunk, no work for us. Typically a `style.css` from cache group.
```

Comment:
I was using the generated asset.php files for standalone CSS files output from Webpack to provide a version number, e.g.:

```php
$componentAndHookStylesMetadata =
	require MY_DIR . '/build/component-and-hook-styles.asset.php';

wp_enqueue_style(
	'my-plugin-component-and-hook-styles',
	plugin_dir_url(__DIR__) . 'build/component-and-hook-styles.css',
	[],
	$componentAndHookStylesMetadata['version']
);
```

It seems the changes in this PR have broken this, resulting in fatal errors when my plugin tries to load the asset.php file that is no longer created.

I've worked around this by changing the code to:

```php
$lastModifiedTimestamp =
	filemtime(plugin_dir_path(__DIR__) . 'build/component-and-hook-styles.css');

// (DT::now() is just a custom util to get a DateTimeImmutable for the current moment.)
if ($lastModifiedTimestamp === false) $lastModifiedTimestamp = DT::now()->getTimestamp();

wp_enqueue_style(
	'my-plugin-component-and-hook-styles',
	plugin_dir_url(__DIR__) . 'build/component-and-hook-styles.css',
	[],
	(string) $lastModifiedTimestamp,
);
```

Maybe this is how I should've been handling it from the start, but I figured I mention that this did indeed affect at least one person.

Reply:
@ZebulanStanphill, how would you get the asset file generated for the CSS file? Was it set explicitly as an entry point?

It's also possible to use the `version` field in the `block.json` file, so it gets automatically used with CSS files:
https://github.com/WordPress/gutenberg/blob/trunk/docs/reference-guides/block-api/block-metadata.md#version

The solution you have works great but only in case when your block is always loaded from the same server.

Reply:
@gziolo Yeah, I had an SCSS file as one of my Webpack entry points, and I used [`mini-css-extract-plugin`](https://webpack.js.org/plugins/mini-css-extract-plugin/) to put the styles in a CSS file, and [`webpack-remove-empty-scripts`](https://www.npmjs.com/package/webpack-remove-empty-scripts) to remove the empty `.js` file generated from the entry point.

Thanks for the advice on using a `version` field on the block.json. The project I'm working on is internal and only used on a single site and server anyway, though, so I suppose it's fine to continue using `filemtime` in my case. (And also, I'd probably forget to update the `version` field on the blocks if I was using that, haha.)

Reply:
I would expect that updating `version` is easy to miss, so maybe we should figure out how to generate asset files also for all CSS files after all. It definitely didn't produce asset files for CSS files extracted when importing them from JS. We never explored that because in CSS, unlike in JavaScript, we don't need to sort out their dependent styles.